src/clr-angular/popover/dropdown/providers/dropdown-focus-handler.service.ts

```diff
@@ -109,11 +99,26 @@ export class DropdownFocusHandler implements FocusableItem {
   }
   set container(el: HTMLElement) {
     this._container = el;
-    if (!this.parent) {
+
+    // whether root container or not, tab key should always toggle (i.e. close) the container
+    this.renderer.listen(el, 'keydown.tab', event => this.ifOpenService.toggleWithEvent(event));
+
+    if (this.parent) {
+      // if it's a nested container, pressing esc has the same effect as pressing left key, which closes the current
+      // popup and moves up to its parent. Here, we preventDefault and stop propagation so that the parent container
+      // doesn't receive the esc keydown
+      this.renderer.listen(el, 'keydown.esc', event => {
+        this.focusService.move(ArrowKeyDirection.LEFT, event);
```

Comment:
I think you are right. Removing preventDefault.

As for your other comment, I think this is ok to stop propagation because the dropdown is a complex ARIA-widget for which we are defining and providing expected keyboard navigation behavior.

Reply:
The case where it could be problematic is that when a user opens our dropdown menu and something like dialog, modal, or toast pops up asynchronously that normally gets closed by global `esc` event would be interfered with this stopPropagation.